optimize/minimize.go

```diff
@@ -35,10 +35,21 @@ type Task struct {
 
 // Location represents a location in the optimization procedure.
 type Location struct {
-	X        []float64
-	F        float64
+	// X is the function input for the location.
+	X []float64
+	// F is the result of evaluating the function at X.
+	F float64
+	// Gradient holds the gradient of the function at X.
+	// The length of Gradient must match the length of X
+	// or be zero. If the capacity of Gradient is less
+	// than the length of X, a new slice will be allocated.
 	Gradient []float64
-	Hessian  *mat.SymDense
+	// Hessian holds the curvature of the function at X.
```

Comment:
I would avoid mentioning curvature, there is a relation but it is not that the Hessian is the curvature.

Reply:
What would you prefer?

Reply:
I would just say "Hessian holds the Hessian of the function at X". That's 5 Hessians on 3 lines but maybe it doesn't matter.

Reply:
Mmm. I was hoping for something a little more descriptive. "Hessian holds the second-order partial derivatives of the function at X."?

Reply:
Yeah, that's fine too and better than curvature. Then also gradient could be equally documented as "Gradient holds the first-order partial derivatives of the function at X".

Reply:
Done. PTAL